modules/ocf/manifests/systemuser.pp

```diff
@@ -0,0 +1,20 @@
+define ocf::systemuser(
+  $ensure = present,
+  $opts = {},
+){
+
+  $groups_real = $opts['groups'] ? {
+    Array   => $opts['groups'] + ['sys'],
+    String  => [$opts['groups'], 'sys'],
+    default => ['sys'],
+  }
+  $updated = { 'groups' =>  $groups_real }
+  # Rightmost hash takes precedence on merge
+  $updated_opts = $opts + $updated
+
+  user { $title:
+    ensure => $ensure,
+    system => true,
+    *      => $opts,
```

Comment:
Should this be `updated_opts`?

Reply:
This breaks when `ensure` or `system` are also passed in as part of `$opts` with something like `Server Error: Evaluation Error: The attribute 'ensure' has already been set`, even if the passed in values are the same as the ones already specified here.

Can the `ensure` just be passed directly through instead of special-cased and `system` validated with an `assert` at the top of the module and some error raised if it's set? (that or removing it from the opts passed through with `$opts - 'system'` should also work, but then it'll fail silently, which I think is probably worse for something like this which is essentially operator error when it does happen)